tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
-}
+	/* Enable newly created sink */
+	isoal_sink_enable(sink_hdl);
 
+	return sink_hdl;
+}
 
 /**
- * @brief   Test unframed single PDU in a single SDU, PDU with error
+ * Initialize the given test data buffer with a ramp pattern
+ * @param buf  Test data buffer pointer
+ * @param size Length of test data
  */
-void test_unframed_single_pdu_err(void)
+static void init_test_data_buffer(uint8_t *buf, uint16_t size)
 {
-	clear_sdu_buf();
+	zassert_not_null(buf, "");
 
-	pdu_ref = &testdata[0];
-	construct_pdu_unframed(PDU_BIS_LLID_COMPLETE_END, pdu_ref, 23, payload_number++, 9249, 1);
-
-	/* Test recombine */
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* Expecting 0 data to be written but with error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_ERRORS, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
+	for (uint16_t i = 0; i < size; i++) {
+		buf[i] = (uint8_t)(i & 0x00FF);
+	}
 }
 
 /**
- * @brief   Trigger payload number sequence error path
+ * Test Suite	:	RX basic test
+ *
+ * Test creating and destroying sinks upto the maximum with randomized
+ * configuration parameters.
  */
-void test_unframed_seq_err(void)
+ZTEST(test_rx_basics, test_sink_create_destroy)
 {
-	clear_sdu_buf();
-
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, &testdata[0], 3,
-			       payload_number++, 10000, 0);
-	/* Test recombine */
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	zassert_equal(sink->sdu_production.sdu_written, 3, "written=%u",
-		      sink->sdu_production.sdu_written);
-
-	construct_pdu_unframed(PDU_BIS_LLID_COMPLETE_END, &testdata[3], 7, 123123, 20000, 0);
-	/* Test recombine, should now trigger emit since this is last PDU in SDU */
-	pdu_ref = &testdata[0];
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	/* Expecting data to be written but with error status */
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_LOST_DATA,
-		      "sdu_status=0x%x", sink->sdu_production.sdu_status);
+	isoal_sink_handle_t sink_hdl[CONFIG_BT_CTLR_ISOAL_SINKS];
+	uint32_t stream_sync_delay;
+	uint32_t group_sync_delay;
+	uint8_t sdu_interval_int;
+	uint8_t iso_interval_int;
+	uint8_t flush_timeout;
+	uint32_t iso_interval;
+	uint32_t sdu_interval;
+	uint8_t burst_number;
+	uint8_t pdus_per_sdu;
+	isoal_status_t res;
+	uint16_t handle;
+	int32_t latency;
+	bool framed;
+
+	res = isoal_init();
+	zassert_equal(res, ISOAL_STATUS_OK, "res=0x%02x", res);
+
+	res = isoal_reset();
+	zassert_equal(res, ISOAL_STATUS_OK, "res=0x%02x", res);
+
+	for (int role = 0; role <= 3; role++) {
```

Comment:
Where is `role` coming from? Global variable?

Reply:
conn.h:
`
BT_CONN_ROLE_CENTRAL = 0
BT_CONN_ROLE_PERIPHERAL = 1
`

isoal.h:
`BT_ROLE_BROADCAST (BT_CONN_ROLE_PERIPHERAL + 1)`

3 is undefined as an error case.

Reply:
Fair. To me it would make more sense to store those role in an array, and use ARRAY_SIZE when iterating over them :) But it is OK as is.